libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java

```diff
@@ -159,76 +156,73 @@ private boolean launchWithRoute(NavigationViewOptions options) {
   private void extractRouteFromOptions(NavigationViewOptions options) {
     DirectionsRoute route = options.directionsRoute();
     if (route != null) {
-      cacheRouteProfile(options, route);
+      cacheRouteOptions(route.routeOptions());
+      cacheRouteProfile(options);
       cacheRouteLanguage(options, route);
-      cacheRouteDestination(route);
       this.route.setValue(route);
     }
   }
 
+  /**
+   * Extracts the {@link Point} coordinates, adds a destination marker,
+   * and fetches a route with the coordinates.
+   *
+   * @param options containing origin and destination
+   */
+  private void extractCoordinatesFromOptions(NavigationViewOptions options) {
+    if (options.origin() != null && options.destination() != null) {
+      cacheRouteProfile(options);
+      cacheRouteLanguage(options, null);
+      Point origin = options.origin();
+      destination.setValue(options.destination());
+      fetchRouteFromCoordinates(origin, destination.getValue());
+    }
+  }
+
+  private void cacheRouteOptions(RouteOptions routeOptions) {
```

Comment:
Does this really need it's own method? I'm not sure `cacheRouteDestination()` fits under the name of this method either